Source/WebKit/UIProcess/API/Cocoa/WKWebView.mm

```diff
@@ -1954,9 +1954,10 @@ - (void)createPDFWithConfiguration:(WKPDFConfiguration *)pdfConfiguration comple
     if (pdfConfiguration && !CGRectIsNull(pdfConfiguration.rect))
         floatRect = WebCore::FloatRect(pdfConfiguration.rect);
 
+    auto drawToPDFFunction = _page->preferences().siteIsolationEnabled() ? &WebKit::WebPageProxy::drawCompositedToPDF : &WebKit::WebPageProxy::drawToPDF;
```

Comment:
Should we consider a separate setting for this, and enable it by default (or if site isolation is enabled).

The idea would be to get people using the new path and get it tested before site-isolation ships (with the option to disable it quickly if things go wrong).

Reply:
I agree that testing this part should be separate from the site isolation feature. I will socialize this topic to get an agreement and see what the proper name for this feature should be.

Reply:
I will submit this patch using the `siteIsolationEnabled` setting for now. And I may submit another patch for this composited snapshotting later.

Reply:
I will remove the iOS changes also because I realized it is more complicated than macOS and it may require snapshotting one PDF page from the web page. Or it may require all the PDF pages. And it may work on a background thread or the main thread. This makes the spans hotting work sometimes synchronously.